### PR TITLE
RavenDB-14232: Better error when failed to update expired license on …

### PR DESCRIPTION
**Pull request by garayx**
…registration

### CODE REVIEW

File: src/Raven.Server/Commercial/LicenseManager.cs [hunk 1]
```diff
@@ -537,9 +537,15 @@ private async Task<bool> ContinueActivatingLicense(License license, bool skipLea
...
                     SetLicense(license.Id, newLicenseStatus.Attributes);
-
-                    // license expired, we'll try to update it
-                    license = await GetUpdatedLicenseInternal(license);
+                    try
+                    {
+                        // license expired, we'll try to update it
+                        license = await GetUpdatedLicenseInternal(license);
+                    }
+                    catch (Exception e)
```
Comment (grisha-kotler) [inline src/Raven.Server/Commercial/LicenseManager.cs:545]: `HttpRequestException`?

File: src/Raven.Server/Commercial/LicenseManager.cs [hunk 2]
```diff
@@ -537,9 +537,18 @@ private async Task<bool> ContinueActivatingLicense(License license, bool skipLea
...
+                    {
+                        if (e is HttpRequestException)
```
Comment (ayende) [inline src/Raven.Server/Commercial/LicenseManager.cs:547]: Include IOException / SocketException here as well?

Comment (ppekrol) [reply src/Raven.Server/Commercial/LicenseManager.cs:547]: Probably not needed? https://docs.microsoft.com/en-us/dotnet/api/system.net.http.httpclient.sendasync?view=netcore-3.1#System_Net_Http_HttpClient_SendAsync_System_Net_Http_HttpRequestMessage_ but maybe we should include `TaskCanceledException` ?

Comment (ayende) [reply src/Raven.Server/Commercial/LicenseManager.cs:547]: Absolutely `TaskCanceledException`, but...

```
var client = new HttpClient();
client.GetAsync("http://e.example.com").GetAwaiter().GetResult();
```

Gives:

```
Unhandled exception. System.Net.Http.HttpRequestException: No such host is known.
 ---> System.Net.Sockets.SocketException (11001): No such host is known.
   at System.Net.Http.ConnectHelper.ConnectAsync(String host, Int32 port, CancellationToken cancellationToken)
   --- End of inner exception stack trace ---
```

Comment (ppekrol) [reply src/Raven.Server/Commercial/LicenseManager.cs:547]: Yes, but the 'SocketException' is an inner ex of HttpRequestException?

Comment (ayende) [reply src/Raven.Server/Commercial/LicenseManager.cs:547]: Hm... that is an inner exception, never mind.

Comment (garayx) [reply src/Raven.Server/Commercial/LicenseManager.cs:547]: Do we want such deep error handling ? It still will throw a LicenseExpiredException with inner exception if it is not a HttpRequestException

Comment (ppekrol) [reply src/Raven.Server/Commercial/LicenseManager.cs:547]: @garayx what you did is fine, I think

**Pull request merged**